<launch>	
	<!--This is a launch file for the heron vessel move_base function-->
	<!--Run the map server with a nanyang lake map-->
	<node name="map_server" pkg="map_server" type="map_server" args="$(find heron_nav)/maps/nanyanglake.yaml"/>

	<!--Launch move_base and configuration files-->
	<node name="move_base" pkg="move_base" type="move_base" output="screen">
		<rosparam file="$(find heron_nav)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/>
		<rosparam file="$(find heron_nav)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>
		<rosparam file="$(find heron_nav)/config/local_costmap_params.yaml" command="load"/>
		<rosparam file="$(find heron_nav)/config/global_costmap_params.yaml" command="load"/>
		<rosparam file="$(find heron_nav)/config/base_local_planner_params.yaml" command="load"/>
	</node>

	<!--include the robot_localization launch file-->
	<include file="$(find robotx_control)/launch/wamv_no_vel_r_l_control.launch"/>	
	<!--Run the tf static transform between map and odom-->
	<node name="map_odom_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0  /map /odom 20"/>
	<!--node name="imu_link_imu_broadcaster" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1.0 imu_link imu 100"/-->
	<!--node name="twist_to_helm" pkg="heron_controller" type="twist_Helm"/-->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find heron_nav)/launch/default.rviz"/>
</launch>
